django/core/management/commands/migrate.py

```diff
@@ -259,63 +259,54 @@ def migration_progress_callback(self, action, migration=None, fake=False):
                 self.stdout.write(self.style.SUCCESS(" DONE" + elapsed))
 
     def sync_apps(self, connection, app_labels):
-        "Runs the old syncdb-style operation on a list of app_labels."
-        cursor = connection.cursor()
+        """Run the old syncdb-style operation on a list of app_labels."""
 
-        try:
-            # Get a list of already installed *models* so that references work right.
+        with connection.cursor() as cursor:
             tables = connection.introspection.table_names(cursor)
-            created_models = set()
 
-            # Build the manifest of apps and models that are to be synchronized
-            all_models = [
-                (app_config.label,
-                    router.get_migratable_models(app_config, connection.alias, include_auto_created=False))
-                for app_config in apps.get_app_configs()
-                if app_config.models_module is not None and app_config.label in app_labels
-            ]
+        # Get a list of already installed *models* so that references work right.
+        created_models = set()
 
-            def model_installed(model):
-                opts = model._meta
-                converter = connection.introspection.table_name_converter
-                # Note that if a model is unmanaged we short-circuit and never try to install it
-                return not (
-                    (converter(opts.db_table) in tables) or
-                    (opts.auto_created and converter(opts.auto_created._meta.db_table) in tables)
-                )
+        # Build the manifest of apps and models that are to be synchronized
+        all_models = [
+            (app_config.label,
+                router.get_migratable_models(app_config, connection.alias, include_auto_created=False))
+            for app_config in apps.get_app_configs()
+            if app_config.models_module is not None and app_config.label in app_labels
+        ]
 
-            manifest = OrderedDict(
```

Comment:
this should probably have a comment that the editor performs the deferred SQL upon successful exit of context manager, took me a bit of time to track that down